aquadoggo/src/db/stores/document.rs

```diff
@@ -459,14 +464,59 @@ async fn insert_document_view(
             )
         VALUES
             ($1, $2, $3)
-        ON CONFLICT(document_view_id) DO NOTHING -- @TODO: temp fix for double document view insertions: https://github.com/p2panda/aquadoggo/issues/398
         ",
     )
     .bind(document_view.id().to_string())
     .bind(document_id.to_string())
     .bind(schema_id.to_string())
-    .execute(tx)
-    .await
+    .execute(&mut *tx)
+    .await;
+
+    match result {
+        Ok(_) => Ok(true),
+        Err(err) => match err {
+            // Handle errors returned from the database
+            sqlx::Error::Database(_) => {
```

Comment:
Can we make a query if that view exists before we insert, instead of handling an error?

I can see how the error approach makes one request less in case no data was in the database but it looks like it tries to insert on _all_ database errors?

I'm not against it, just want to compare the approaches.